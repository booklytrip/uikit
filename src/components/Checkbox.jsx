/**
 * A checkbox allows a user to select a value from a small set of options,
 * often binary.
 *
 * @flow
 */

import React from 'react';
import classNames from 'classnames';

export type CheckboxProps = {
    style: Object,
    radio?: boolean,
    checked?: boolean,
    disabled?: boolean,
    toggle?: boolean,
    label?: string | Object,
    value: string,
    name: string,
    onChange: Function,
};

const styles = {
    checkbox: {
        display: 'flex',
    },
    label: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
    },
};

const Checkbox = (
    {
        style,
        radio,
        checked,
        disabled,
        toggle,
        label,
        value,
        name,
        onChange,
    }: CheckboxProps,
) => (
    <div
        className={classNames('ui checkbox', {
            radio,
            toggle,
            disabled,
        })}
        style={Object.assign({}, styles.checkbox, style)}
    >
        <input
            type={radio ? 'radio' : 'checkbox'}
            id={`${name}-${value}`}
            name={name}
            value={value}
            checked={checked}
            onChange={e => onChange(radio ? value : !checked)}
        />
        {label &&
            <label style={styles.label} htmlFor={`${name}-${value}`}>
                {label}
            </label>}
    </div>
);

export default Checkbox;
