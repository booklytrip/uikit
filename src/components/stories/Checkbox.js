import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import centered from '@kadira/react-storybook-decorator-centered';

import Checkbox from '../Checkbox.jsx';

storiesOf('Checkbox', module)
    .addDecorator(centered)
    .add('default', () => (
        <Checkbox
            label="Lorem ipsum dolor sit amet"
            onChange={action('onChange')}
        />
    ))
    .add('disabled', () => (
        <Checkbox
            label="Lorem ipsum dolor sit amet"
            onChange={action('onChange')}
            disabled
        />
    ))
    .add('checked', () => (
        <Checkbox
            label="Lorem ipsum dolor sit amet"
            onChange={action('onChange')}
            checked
        />
    ))
    .add('toggle', () => (
        <Checkbox
            label="Lorem ipsum dolor sit amet"
            onChange={action('onChange')}
            toggle
        />
    ));
