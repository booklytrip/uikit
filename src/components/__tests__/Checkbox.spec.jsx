import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Checkbox from '../Checkbox';

describe('Checbox', () => {
    it('renders without crashing', () => {
        mount(<Checkbox />);
    });

    it('should call onChange event if changed', () => {
        const spy = jest.fn();
        const checkbox = mount(<Checkbox onChange={spy} />);
        checkbox.find('input').simulate('change');

        expect(spy).toHaveBeenCalled();
    });

    describe('Snapshots', () => {
        it('renders default checkbox correctly', () => {
            const rendered = renderer.create(<Checkbox />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders radio type checkbox correctly', () => {
            const rendered = renderer.create(<Checkbox radio />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders toogle checkbox correctly', () => {
            const rendered = renderer.create(<Checkbox toggle />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
