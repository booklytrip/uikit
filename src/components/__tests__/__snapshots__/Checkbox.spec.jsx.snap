exports[`Checbox Snapshots renders default checkbox correctly 1`] = `
<div
  className="ui checkbox"
  style={
    Object {
      "display": "flex",
    }
  }>
  <input
    checked={undefined}
    id="undefined-undefined"
    name={undefined}
    onChange={[Function]}
    type="checkbox"
    value={undefined} />
</div>
`;

exports[`Checbox Snapshots renders radio type checkbox correctly 1`] = `
<div
  className="ui checkbox radio"
  style={
    Object {
      "display": "flex",
    }
  }>
  <input
    checked={undefined}
    id="undefined-undefined"
    name={undefined}
    onChange={[Function]}
    type="radio"
    value={undefined} />
</div>
`;

exports[`Checbox Snapshots renders toogle checkbox correctly 1`] = `
<div
  className="ui checkbox toggle"
  style={
    Object {
      "display": "flex",
    }
  }>
  <input
    checked={undefined}
    id="undefined-undefined"
    name={undefined}
    onChange={[Function]}
    type="checkbox"
    value={undefined} />
</div>
`;
